### DISCUSSION TITLE
I have a question about pull.

### DISCUSSION THREAD

**Discussion by jhk01007** [community/community#56023] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

When A and B collaborated with GitHub, A pushed the modified code first, and B tried to pull to combine his modified code and A's modified code, but the pull failed. 
I tried pushing on another branch first and then pulling but still fails. Is there any other solution?

**Comment by jge162** [comment]
Couple options you can do @jhk01007. Update branch name from origin/Master to your working branches. 

- From b branch do a `git fetch` and then `git merge origin/master`. 

- You can even try rebase. `git fetch`
`git rebase origin/master` 

But, remember to check with A to ensure these changes are suitable. Report back.

**Reply by airtower-luna** [reply]
Merging or rebasing are generally the right solutions. `reset --hard` would destroy *all* local changes on the current branch, so better don't do that. :warning:

**Reply by jge162** [reply]
I really didn’t like reset option @airtower-luna I tried to discourage it. Next time I won’t probably mention. Thanks for input.

**Reply by airtower-luna** [reply]
I don't think it's wrong to mention at all, just that it's important to be clear about what it does. It might be possible to recover commits using reflog, but for someone who's trying to figure out how to merge/rebase that'd be a much bigger challenge. :slightly_smiling_face:
